auth/password_authenticator.cc

```diff
@@ -102,8 +100,9 @@ future<> password_authenticator::migrate_legacy_metadata() const {
         return do_for_each(*results, [this](const cql3::untyped_result_set_row& row) {
             auto username = row.get_as<sstring>("username");
             auto salted_hash = row.get_as<sstring>(SALTED_HASH);
+            static const auto query = update_row_query();
```

Comment:
`static`? Doesn't this defeat the point of destaticizing the string inside update_row_query()?

Reply:
it's not dead code at least in a theory, in a follow-up I'll deal with making sure v1 migration code doesn't trigger for v-2 (no harm if it does but not needed), it's better than tagging every function on their own - it's not like they are generic or even remotely-generic functions anyway.